r1cs/gadgets/std/src/fields/nonnative/nonnative_field_gadget.rs

```diff
@@ -633,13 +641,13 @@ impl<SimulationF: PrimeField, ConstraintF: PrimeField>
         //     2 * bits_per_limb + surfeit' <= CAPACITY - 2,
         // ``
         // with `surfeit' = log(num_limbs * (num_add(L) + 1) * (num_add(R) + 1) + num_limbs)`.
-        let params = get_params(SimulationF::size_in_bits(), ConstraintF::size_in_bits());
-        let num_add_bound =  BigUint::from(params.num_limbs) 
+        //ToDo: check this constraint, as by comparison with prereduce constraint a factor of 2 seems to be missing
```

Comment:
@UlrichHaboeck75 what do you think ?

Reply:
I confirm. The factor 2 is missing. The correct formula should be
```
   surfeit' = log(2 * num_limbs * (num_add(L) + 1) * (num_add(R) + 1) + num_limbs)
```

Reply:
Let's remove this TODO now no ?